lab_3/main.py

```diff
@@ -85,7 +153,20 @@ def decode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
     :param corpus: an encoded tuple of sentences
     :return: a tuple of the decoded sentences
     """
-    pass
+
+    if not isinstance(storage, LetterStorage) or not isinstance(corpus, tuple):
+        return ()
+
+    decoded_text = ()
+    for sentence in corpus:
+        word_tuple = ()
+        for word in sentence:
+            letter_tuple = ()
+            for letter_id in word:
+                letter_tuple += (storage.get_id_by_letter(letter_id),)
+            word_tuple += (letter_tuple,)
+        decoded_text += (word_tuple,)
+    return decoded_text
```

Comment:
this function doesn't pass the "ideal" test and i cannot understand why

Reply:
You have to get the letter by id as it is a decoding function. Now it is `get_id_by_letter` but there should be used the `get_letter_by_id` method.